### PR TITLE
C4C-190 - User Authentication in New Web App

### PR DESCRIPTION
**Pull request by codyhaws**
Contains all of the necessary packages and files to enable user authentication in the new .NET Core web app. Some file bloat is acknowledged due to unused features in ASP.NET Identity libraries, leaving them in for now to promote stability. Uses a new MySQL database running on Google Cloud for user storage. If there are any questions, please let me know.

### CODE REVIEW

File: C4CCoreApp/C4CCoreApp/Migrations/ApplicationDbContextModelSnapshot.cs [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+using System;
+
+namespace C4CCoreApp.Migrations
+{
+    [DbContext(typeof(ApplicationDbContext))]
+    partial class ApplicationDbContextModelSnapshot : ModelSnapshot
+    {
+        protected override void BuildModel(ModelBuilder modelBuilder)
+        {
+#pragma warning disable 612, 618
```
Comment (murphyjp) [inline C4CCoreApp/C4CCoreApp/Migrations/ApplicationDbContextModelSnapshot.cs]: I'm sure it's trivial and/or defensible, but as someone who didn't write the code, it'd be nice to see a short comment saying what warning this is and why it's ignored. a simple // clause on the same line would do it

File: C4CCoreApp/C4CCoreApp/Services/EmailSender.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        /// and potentially have all options configurable in the app.
+        /// </summary>
+        /// <param name="address">The e-mail address to use as the "From".</param>
+        /// <param name="password">The password for the account associated with the address.</param>
+        /// <param name="server">The SMTP server to use when sending the e-mail.</param>
+        /// <param name="name">The name to be shown instead of the "From" address in e-mail clients.</param>
+        protected void GetSenderInfo(out string address, out string password, out string server, out string name)
+        {
+            address = "chix4acause.orders@yandex.com";
+            password = "DietPepsi";
```
Comment (murphyjp) [inline C4CCoreApp/C4CCoreApp/Services/EmailSender.cs]: Perhaps a bugfix for a future sprint, but passwords should not be hardcoded, especially in a publicly accessible git repo like this.

Comment (codyhaws) [reply C4CCoreApp/C4CCoreApp/Services/EmailSender.cs]: Yup, we actually have a bug in our current sprint to encrypt all of our API keys, passwords, etc. using Google's Key Management Service, it's what I'll be taking on next.

File: C4CCoreApp/C4CCoreApp/Migrations/ApplicationDbContextModelSnapshot.cs (hunk 1 above)
Comment (codyhaws) [reply C4CCoreApp/C4CCoreApp/Migrations/ApplicationDbContextModelSnapshot.cs]: The files in the Migrations folder are auto-generated by Microsoft's Entity Framework library, so I'm not entirely sure why lines like that one exist. I'll attempt to step through the files and provide what documentation I can with the knowledge I have.

Comment (murphyjp) [reply C4CCoreApp/C4CCoreApp/Migrations/ApplicationDbContextModelSnapshot.cs]: I assumed you had added the warning ignore. If not, nvm! Thanks.

**Comment by codyhaws** [comment]
Heavily updated documentation to assuage earlier concerns expressed by reviewers. Ready to merge.

**Pull request merged**